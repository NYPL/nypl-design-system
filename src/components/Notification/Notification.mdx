import { ArgTypes, Canvas, Description, Meta, Source } from "@storybook/blocks";

import * as NotificationStories from "./Notification.stories";
import Link from "../Link/Link";
import ComponentChangelogTable from "../../utils/ComponentChangelogTable";
import { changelogData } from "./notificationChangelogData";

<Meta of={NotificationStories} />

# Notification

| Component Version | DS Version   |
| ----------------- | ------------ |
| Added             | `0.23.2`     |
| Latest            | `Prerelease` |

## Table of Contents

- {<Link href="#overiew" target="_self">Overview</Link>}
- {<Link href="#component-props" target="_self">Component Props</Link>}
- {<Link href="#accessibility" target="_self">Accessibility</Link>}
- {<Link href="#notification-heading" target="_self">Notification Heading</Link>}
- {<Link href="#notification-icon" target="_self">Notification Icon</Link>}
- {<Link href="#dismissible" target="_self">Dismissible</Link>}
- {<Link href="#changelog" target="_self">Changelog</Link>}

## Overview

<Description of={NotificationStories} />

The `Notification` component is a configurable callout that should be used to
display important messages.

In the preview below, the border around the `Notification` is not part of the
component. It has been added to help illustrate how the `Notification` sits
within a parent element.

## Component Props

<Canvas of={NotificationStories.WithControls} />

<ArgTypes of={NotificationStories.WithControls} />

## Accessibility

The `Notification` component renders with an HTML `aside` element as its
wrapper. This is an HTML landmark element that is similar to adding an attribute
of `role="complementary"`. For accessibility purposes, landmark elements should
not be rendered inside other landmark elements such as the `header` and `footer`
landmark elements. Adding a `Notification` component inside an HTML `main`
landmark element is acceptable.

### ARIA Label

Every `Notification` component must have a unique `aria-label` attribute. This
is set through the `ariaLabel` prop. A unique `aria-label` value, along with the
`aside` HTML landmark element, helps screen readers better navigate a page with
multiple `Notification`s.

### Icons

Icons rendered in the `Notification` component are decorative by default which
means that they are hidden to screen readers. Since the "X" close icon inside
the dismissible button is decorative and because there is no discernible text
inside the button, an `aria-label` attribute is added to the button.

### Language Patterns

When the `Notification` component is rendered, the text within the component
should always include wording that aligns with the type of notitication that is
being delivered. For example, if the notification is a warning or error, the
heading or body copy within the notification should use the words `"warning"` or
`"error"` and provide a clear description of the existing or potential isssue.

### Dynamic Notifications

If dynamically generated notifications are required, a container into which the
notification will be injected should be included in the DOM. The container
element should have `aria-live="polite"` applied. The container element should
always be rendered in the DOM regardless of the presence of the Notification
component. This DOM structure will allow the newly generated content to be
recognized and announced by assistive technology.

<Source
  code={`
<div aria-live="polite">
  <Notification {...props} />
</div>
`}
  language="tsx"
/>

### Resources

- [MDN Aria: complementary role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/complementary_role)
- [Deque Creating Accessible SVGs](https://www.deque.com/blog/creating-accessible-svgs/)
- [CSS Tricks Accessible SVG Icons](https://css-tricks.com/accessible-svg-icons/)

## Notification Heading

The `Notification` component can be rendered with and without a heading. By
default, when a title is passed to the `notificationHeading` prop, the
`Notification` component will render an `h4` element. If this causes accessibility
issues in your app, you can pass in a custom `Heading` element.

<Canvas of={NotificationStories.NotificationHeading} />

In the following examples, the `Notification` component will render
custom `h3` and `h5` elements:

<Source
  code={`

const customH3 = <Heading level="h3">Custom H3 Heading</Heading>;
const customH5 = <Heading level="h5">Custom H5 Heading</Heading>;

<Notification notificationHeading={customH3} {...props} />
<Notification notificationHeading={customH5} {...props} />
`} language="tsx" />

<Canvas of={NotificationStories.CustomNotificationHeading} />

## Notification Icon

### Default Icon

The `Notification` icon can be displayed with or without an icon. The icon can
be hidden with the `showIcon` prop set to `false`.

<Canvas of={NotificationStories.NotificationIcon} />

### Custom Icon

The default icon can be overridden by using the `icon` prop to pass a custom `Icon` component.

<Canvas of={NotificationStories.CustomIcon} />

## With HTML Content

The `notificationContent` prop can accept HTML.

<Canvas of={NotificationStories.HTMLContent} />

## Dismissible

A dismissible `Notification` component can be created by setting the
`dismissible` prop to `true`. Once the "X" close icon on the upper right is
clicked, the `Notification` will be removed from the DOM, therefore it only
appears once.

<Canvas of={NotificationStories.Dismissible} />

## Changelog

<ComponentChangelogTable changelogData={changelogData} />
