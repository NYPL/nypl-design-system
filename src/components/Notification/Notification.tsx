import { Box, chakra, useMultiStyleConfig } from "@chakra-ui/react";
import React, { useState } from "react";

import Button from "../Button/Button";
import Heading from "../Heading/Heading";
import Icon, { IconColors, IconNames, IconSizes } from "../Icons/Icon";

export type NotificationTypes = "standard" | "announcement" | "warning";
interface BaseProps {
  /** Optional prop to control text alignment in `NotificationContent` */
  alignText?: boolean;
  /** Optional custom `Icon` that will override the default `Icon`. */
  icon?: JSX.Element;
  /** ID that other components can cross reference for accessibility purposes. */
  id?: string;
  /** Optional prop to control horizontal alignment of the `Notification` content */
  isCentered?: boolean;
  /** Optional prop to control the coloring of the `Notification` text and the
   * visibility of an applicable icon. */
  notificationType?: NotificationTypes;
}

// Used for `NotificationHeading` and `Notification`
type BasePropsWithoutAlignText = Omit<BaseProps, "alignText">;
// Used for `NotificationContent`
type BasePropsWithoutIsCentered = Omit<BaseProps, "isCentered">;

export interface NotificationProps extends BasePropsWithoutAlignText {
  /** Label used to describe the `Notification`'s aside HTML element. */
  ariaLabel?: string;
  /** Additional `className` to add.  */
  className?: string;
  /** Optional prop to control whether a `Notification` can be dismissed
   * (closed) by a user. */
  dismissible?: boolean;
  /** ID that other components can cross reference for accessibility purposes. */
  id?: string;
  /** Optional custom `Icon` that will override the default `Icon`. */
  icon?: JSX.Element;
  /** Optional prop to control the margin around the `Notification` component. */
  noMargin?: boolean;
  /** Content to be rendered in a `NotificationContent` component. */
  notificationContent: string | JSX.Element;
  /** Content to be rendered in a `NotificationHeading` component. */
  notificationHeading?: string;
  /** Prop to display the `Notification` icon. Defaults to `true`. */
  showIcon?: boolean;
}

/**
 * NotificationHeading child-component.
 */
export const NotificationHeading = chakra(
  (props: React.PropsWithChildren<BasePropsWithoutAlignText>) => {
    const { children, icon, id, isCentered, notificationType, ...rest } = props;
    const styles = useMultiStyleConfig("NotificationHeading", {
      icon,
      isCentered,
      notificationType,
    });
    return (
      <Box as="header" __css={styles} {...rest}>
        {icon}
        <Heading id={`${id}-heading`} level="four" __css={styles.heading}>
          {children}
        </Heading>
      </Box>
    );
  }
);

/**
 * NotificationContent child-component.
 */
export const NotificationContent = chakra(
  (props: React.PropsWithChildren<BasePropsWithoutIsCentered>) => {
    const { alignText, children, icon, notificationType, ...rest } = props;
    const styles = useMultiStyleConfig("NotificationContent", {
      alignText,
      icon,
      notificationType,
    });
    return (
      <Box __css={styles} {...rest}>
        {icon}
        <Box __css={styles.content}>{children}</Box>
      </Box>
    );
  }
);

/**
 * Component used to present users with three different levels of notifications:
 * standard, announcement, and warning.
 */
export const Notification = chakra((props: NotificationProps) => {
  const {
    ariaLabel,
    className,
    dismissible = false,
    icon,
    id,
    isCentered = false,
    noMargin = false,
    notificationContent,
    notificationHeading,
    notificationType = "standard",
    showIcon = true,
    ...rest
  } = props;
  const [isOpen, setIsOpen] = useState(true);
  const handleClose = () => setIsOpen(false);
  const styles = useMultiStyleConfig("Notification", {
    dismissible,
    isCentered,
    noMargin,
    notificationType,
  });
  const iconElement = () => {
    const baseIconProps = {
      size: "large" as IconSizes,
      __css: styles.icon,
    };
    // If the icon should not display, return undefined.
    if (!showIcon) {
      return undefined;
    }
    // If a custom icon is passed, add specific `Notification` styles.
    if (icon)
      return React.cloneElement(icon, {
        id: `${id}-custom-notification-icon`,
        ...baseIconProps,
      });
    interface IconProps {
      color: IconColors;
      name: IconNames;
      title: string;
    }
    const iconProps = {
      announcement: {
        color: "section.research.secondary",
        name: "speakerNotes",
        title: "Notification announcement icon",
      } as IconProps,
      standard: {
        color: "ui.black",
        name: "alertNotificationImportant",
        title: "Notification standard icon",
      } as IconProps,
      warning: {
        color: "brand.primary",
        name: "errorFilled",
        title: "Notification warning icon",
      } as IconProps,
    };
    return (
      <Icon
        id={`${id}-notification-icon`}
        {...iconProps[notificationType]}
        {...baseIconProps}
      />
    );
  };
  const dismissibleButton = dismissible && (
    <Button
      aria-label="Close the notification"
      buttonType="link"
      id={`${id}-notification-dismissible-button`}
      onClick={handleClose}
      __css={styles.dismissibleButton}
    >
      <Icon
        id={`${id}-dismissible-notification-icon`}
        name="close"
        size="large"
        title="Notification close icon"
      />
    </Button>
  );
  const iconElem = iconElement();
  const childHeading = notificationHeading && (
    <NotificationHeading
      icon={iconElem}
      id={id}
      isCentered={isCentered}
      notificationType={notificationType}
    >
      {notificationHeading}
    </NotificationHeading>
  );
  // Specific alignment styles for the content.
  const alignText = !!(childHeading && showIcon && (!!icon || !isCentered));
  const childContent = (
    <NotificationContent
      alignText={alignText}
      icon={!childHeading ? iconElem : undefined}
      notificationType={notificationType}
    >
      {notificationContent}
    </NotificationContent>
  );

  // If the `Notification` is closed, don't render anything.
  if (!isOpen) {
    return null;
  }
  return (
    <Box
      aria-label={ariaLabel}
      as="aside"
      className={className}
      data-type={notificationType}
      id={id}
      __css={styles}
      {...rest}
    >
      <Box __css={styles.container}>
        {childHeading}
        {childContent}
      </Box>
      {dismissibleButton}
    </Box>
  );
});

export default Notification;
