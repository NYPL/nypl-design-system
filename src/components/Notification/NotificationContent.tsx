import React from "react";

export interface NotificationProps {}

export default function NotificationContent(
  props: React.PropsWithChildren<NotificationProps>
) {
  const { children } = props;

  return <div className="content">{children}</div>;
}
