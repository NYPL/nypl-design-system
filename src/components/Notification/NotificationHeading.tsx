import React from "react";
import Heading from "../Heading/Heading";

export interface NotificationProps {}

export default function NotificationHeading(
  props: React.PropsWithChildren<NotificationProps>
) {
  const { children } = props;

  // const notificationModifiers = modifiers ? modifiers : [];
  // if (!notificationModifiers.find(modifier => modifier === notificationType)) {
  //   notificationModifiers.push(notificationType);
  // }

  return <Heading level={4}>{children}</Heading>;
}
