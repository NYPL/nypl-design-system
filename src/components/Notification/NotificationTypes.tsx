export enum NotificationTypes {
  Standard = "standard",
  Announcement = "announcement",
  Warning = "warning",
}
