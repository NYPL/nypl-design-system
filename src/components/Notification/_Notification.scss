.notification {
  border-radius: 4px;
  display: flex;
  font-size: var(--font-size--1);
  margin: var(--space-s);
  padding: var(--space) var(--space-l);
  position: relative;

  &__container {
    margin: auto;
    max-width: 1280px;
  }

  .content {
    display: flex;
    justify-content: center;
  }

  &--centered {
    text-align: center;

    header {
      justify-content: center;
    }

    .content {
      display: block;
    }
  }

  &--fill-space {
    border-radius: 0;
    margin: 0;
  }

  &--standard {
    background-color: var(--ui-status-primary);
  }

  &--announcement {
    background-color: var(--ui-gray-light);

    .notification-heading {
      color: var(--section-research-secondary);
    }
  }

  &--warning {
    background-color: var(--ui-gray-xxlight);

    .notification-heading {
      color: var(--brand-primary);
    }

    .notification-content {
      color: var(--brand-primary);
    }
  }

  &--align-text {
    .notification-content {
      padding-left: calc(var(--space-m) + var(--space-s));
    }
  }

  header {
    display: flex;
    margin-bottom: var(--space-s);

    .notification-heading {
      margin-bottom: 0;
    }
  }

  .notification-icon {
    flex-shrink: 0;
    margin-right: var(--space-s);
  }

  .dismissible-button {
    color: var(--ui-black);
    padding: 0;
    position: absolute;
    right: var(--space-xs);
    top: var(--space-xs);
  }
}
