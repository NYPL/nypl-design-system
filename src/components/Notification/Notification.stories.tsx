import * as React from "react";
import { Story } from "@storybook/react";

import Notification, { NotificationProps } from "./Notification";
import { NotificationTypes } from "./NotificationTypes";
import { AccordionAsFAQSet } from "../Accordion/Accordion.stories";

export default {
  title: "Notification",
  component: Notification,
};

const Template: Story<NotificationProps> = args => <Notification {...args} />;

const figmaParameter = {
  design: {
    type: "figma",
    url:
      "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Main?node-id=10766%3A142",
  },
};

const commonArgTypes = {
  placement: {
    defaultValue: "sticky",
  },
};

export const Default = Template.bind({});
Default.args = {
  children: "Notification content",
};
Default.argTypes = {
  notificationType: {
    defaultValue: "standard",
  },
  ...commonArgTypes,
};
Default.storyName = "Default Notification";
Default.parameters = figmaParameter;

export const Announcement = Template.bind({});
Announcement.storyName = "Announcement with Heading";
Announcement.args = {
  notificationType: NotificationTypes.Announcement,
  heading: "New Announcement",
  children:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
};
Announcement.argTypes = commonArgTypes;
Announcement.parameters = figmaParameter;

export const Warning = Template.bind({});
Warning.storyName = "Warning Notification with link";
Warning.args = {
  notificationType: NotificationTypes.Warning,
  children: (
    <>
      Lorem ipsum dolor sit amet, <a href="/">consectetur adipiscing elit</a>,
      sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem
      ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
      incididunt ut labore et dolore magna aliqua.
    </>
  ),
};
Warning.argTypes = commonArgTypes;
Warning.parameters = figmaParameter;

export const DismissibleAnnouncement = Template.bind({});
DismissibleAnnouncement.storyName = "Dismissible Notification";
DismissibleAnnouncement.args = {
  notificationType: "announcement",
  dismissible: true,
  children:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
};
DismissibleAnnouncement.parameters = figmaParameter;
DismissibleAnnouncement.argTypes = commonArgTypes;

export const NotificationInContext = () => (
  <div style={{ height: "300px" }}>
    <Notification>Notification Content</Notification>
    <AccordionAsFAQSet {...AccordionAsFAQSet.args} count={10} />
  </div>
);

NotificationInContext.parameters = figmaParameter;
