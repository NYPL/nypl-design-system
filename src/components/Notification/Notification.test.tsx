import * as React from "react";
import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import renderer from "react-test-renderer";

import Notification from "./Notification";
import { NotificationTypes } from "./NotificationTypes";
import Icon from "../Icons/Icon";
import { IconNames, IconColors, IconSizes } from "../Icons/IconTypes";

describe("Notification Accessibility", () => {
  it("passes axe accessibility test with heading", async () => {
    const { container } = render(
      <Notification
        id="notificationID"
        notificationContent={<>Notification content.</>}
        notificationHeading="Notification Heading"
      />
    );
    expect(await axe(container)).toHaveNoViolations();
  });

  it("passes axe accessibility test without heading", async () => {
    const { container } = render(
      <Notification
        id="notificationID"
        notificationContent={<>Notification content.</>}
      />
    );
    expect(await axe(container)).toHaveNoViolations();
  });

  it("passes axe accessibility test without an icon", async () => {
    const { container } = render(
      <Notification
        id="notificationID"
        notificationContent={<>Notification content.</>}
        notificationHeading="Notification Heading"
        showIcon={false}
      />
    );
    expect(await axe(container)).toHaveNoViolations();
  });

  it("passes axe accessibility test for the dismissible type", async () => {
    const { container } = render(
      <Notification
        dismissible
        id="notificationID"
        notificationContent={<>Notification content.</>}
        notificationHeading="Notification Heading"
      />
    );
    expect(await axe(container)).toHaveNoViolations();
  });
});

describe("Notification", () => {
  let utils;
  beforeEach(() => {
    utils = render(
      <Notification
        ariaLabel="Notification label"
        id="notificationID"
        notificationContent={<>Notification content.</>}
        notificationHeading="Notification Heading"
      />
    );
  });

  it("renders Notification heading child component", () => {
    expect(screen.getByText("Notification Heading")).toBeInTheDocument();
  });

  it("renders Notification content child component", () => {
    expect(screen.getByText("Notification content.")).toBeInTheDocument();
  });

  it("renders with an Icon", () => {
    // The Icon's role is "img".
    expect(screen.queryByRole("img")).toBeInTheDocument();
  });

  it("does not render an Icon", () => {
    utils.rerender(
      <Notification
        id="notificationID"
        notificationContent={<>Notification content.</>}
        notificationHeading="Notification Heading"
        showIcon={false}
      />
    );
    // The Icon's role is "img".
    expect(screen.queryByRole("img")).not.toBeInTheDocument();
  });

  it("renders a custom Icon component", () => {
    utils.rerender(
      <Notification
        icon={
          <Icon
            id="custom-icon"
            className="custom-icon"
            color={IconColors.BrandPrimary}
            name={IconNames.Check}
            size={IconSizes.Large}
          />
        }
        id="notificationID"
        notificationContent={<>Notification content.</>}
        notificationHeading="Notification Heading"
      />
    );
    expect(utils.container.querySelector(".custom-icon")).toBeInTheDocument();
  });

  it("renders the announcement Notification type", () => {
    utils.rerender(
      <Notification
        id="notificationID"
        notificationContent={<>Notification content.</>}
        notificationHeading="Notification Heading"
        notificationType={NotificationTypes.Announcement}
      />
    );

    expect(utils.container.querySelector("aside")).toHaveAttribute(
      "data-type",
      "announcement"
    );
  });

  it("renders the warning Notification type", () => {
    utils.rerender(
      <Notification
        id="notificationID"
        notificationContent={<>Notification content.</>}
        notificationHeading="Notification Heading"
        notificationType={NotificationTypes.Warning}
      />
    );

    expect(utils.container.querySelector("aside")).toHaveAttribute(
      "data-type",
      "warning"
    );
  });

  it("renders with an aria-label attribute", () => {
    expect(screen.getByRole("complementary")).toHaveAttribute(
      "aria-label",
      "Notification label"
    );
  });

  it("renders a dismissible icon", () => {
    utils.rerender(
      <Notification
        dismissible
        id="notificationID"
        notificationContent={<>Notification content.</>}
        notificationHeading="Notification Heading"
        notificationType={NotificationTypes.Standard}
      />
    );
    const icons = screen.queryAllByRole("img");

    expect(icons).toHaveLength(2);
    expect(screen.getByTitle("Notification standard icon")).toBeInTheDocument();
    expect(screen.getByTitle("Notification close icon")).toBeInTheDocument();
  });

  it.skip("renders the UI snapshot correctly", () => {
    const standard = renderer
      .create(
        <Notification
          id="notificationID1"
          notificationContent={<>Notification content.</>}
          notificationHeading="Notification Heading"
        />
      )
      .toJSON();
    const announcement = renderer
      .create(
        <Notification
          id="notificationID2"
          notificationContent={<>Notification content.</>}
          notificationHeading="Notification Heading"
          notificationType={NotificationTypes.Announcement}
        />
      )
      .toJSON();
    const warning = renderer
      .create(
        <Notification
          id="notificationID3"
          notificationContent={<>Notification content.</>}
          notificationHeading="Notification Heading"
          notificationType={NotificationTypes.Warning}
        />
      )
      .toJSON();
    const withoutHeading = renderer
      .create(
        <Notification
          id="notificationID4"
          notificationContent={<>Notification content.</>}
        />
      )
      .toJSON();
    const withoutAnIcon = renderer
      .create(
        <Notification
          id="notificationID5"
          notificationHeading="Notification Heading"
          notificationContent={<>Notification content.</>}
          showIcon={false}
        />
      )
      .toJSON();
    const withoutHeadingAndIcon = renderer
      .create(
        <Notification
          id="notificationID6"
          notificationContent={<>Notification content.</>}
          showIcon={false}
        />
      )
      .toJSON();
    const dismissible = renderer
      .create(
        <Notification
          dismissible
          id="notificationID7"
          notificationContent={<>Notification content.</>}
        />
      )
      .toJSON();

    expect(standard).toMatchSnapshot();
    expect(announcement).toMatchSnapshot();
    expect(warning).toMatchSnapshot();
    expect(withoutHeading).toMatchSnapshot();
    expect(withoutAnIcon).toMatchSnapshot();
    expect(withoutHeadingAndIcon).toMatchSnapshot();
    expect(dismissible).toMatchSnapshot();
  });
});
