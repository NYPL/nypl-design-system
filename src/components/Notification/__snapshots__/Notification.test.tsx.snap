// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders the UI snapshot correctly 1`] = `
<aside
  className="css-1xdhyk6"
  data-type="standard"
  id="notificationID1"
>
  <div
    className="css-0"
  >
    <header
      className="css-1xdhyk6"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon notification-icon css-q3eg3a"
        focusable={false}
        id="notificationID1-notification-icon"
        role="img"
        title="Notification standard icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      <h4
        className="chakra-heading css-1xdhyk6"
        id="notificationID1-heading"
      >
        Notification Heading
      </h4>
    </header>
    <div
      className="css-1xdhyk6"
    >
      <div
        className="css-0"
      >
        Notification content.
      </div>
    </div>
  </div>
</aside>
`;

exports[`Notification renders the UI snapshot correctly 2`] = `
<aside
  className="css-1xdhyk6"
  data-type="announcement"
  id="notificationID2"
>
  <div
    className="css-0"
  >
    <header
      className="css-1xdhyk6"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon notification-icon css-u7u798"
        focusable={false}
        id="notificationID2-notification-icon"
        role="img"
        title="Notification announcement icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      <h4
        className="chakra-heading css-1xdhyk6"
        id="notificationID2-heading"
      >
        Notification Heading
      </h4>
    </header>
    <div
      className="css-1xdhyk6"
    >
      <div
        className="css-0"
      >
        Notification content.
      </div>
    </div>
  </div>
</aside>
`;

exports[`Notification renders the UI snapshot correctly 3`] = `
<aside
  className="css-1xdhyk6"
  data-type="warning"
  id="notificationID3"
>
  <div
    className="css-0"
  >
    <header
      className="css-1xdhyk6"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon notification-icon css-zzgjc6"
        focusable={false}
        id="notificationID3-notification-icon"
        role="img"
        title="Notification warning icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      <h4
        className="chakra-heading css-1xdhyk6"
        id="notificationID3-heading"
      >
        Notification Heading
      </h4>
    </header>
    <div
      className="css-1xdhyk6"
    >
      <div
        className="css-0"
      >
        Notification content.
      </div>
    </div>
  </div>
</aside>
`;

exports[`Notification renders the UI snapshot correctly 4`] = `
<aside
  className="css-1xdhyk6"
  data-type="standard"
  id="notificationID4"
>
  <div
    className="css-0"
  >
    <header
      className="css-1xdhyk6"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon notification-icon css-q3eg3a"
        focusable={false}
        id="notificationID4-notification-icon"
        role="img"
        title="Notification standard icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </header>
    <div
      className="css-1xdhyk6"
    >
      <div
        className="css-0"
      >
        Notification content.
      </div>
    </div>
  </div>
</aside>
`;

exports[`Notification renders the UI snapshot correctly 5`] = `
<aside
  className="css-1xdhyk6"
  data-type="standard"
  id="notificationID5"
>
  <div
    className="css-0"
  >
    <header
      className="css-1xdhyk6"
    >
      <h4
        className="chakra-heading css-1xdhyk6"
        id="notificationID5-heading"
      >
        Notification Heading
      </h4>
    </header>
    <div
      className="css-1xdhyk6"
    >
      <div
        className="css-0"
      >
        Notification content.
      </div>
    </div>
  </div>
</aside>
`;

exports[`Notification renders the UI snapshot correctly 6`] = `
<aside
  className="css-1xdhyk6"
  data-type="standard"
  id="notificationID6"
>
  <div
    className="css-0"
  >
    <div
      className="css-1xdhyk6"
    >
      <div
        className="css-0"
      >
        Notification content.
      </div>
    </div>
  </div>
</aside>
`;

exports[`Notification renders the UI snapshot correctly 7`] = `
<aside
  className="css-1xdhyk6"
  data-type="standard"
  id="notificationID7"
>
  <div
    className="css-0"
  >
    <header
      className="css-1xdhyk6"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon notification-icon css-q3eg3a"
        focusable={false}
        id="notificationID7-notification-icon"
        role="img"
        title="Notification standard icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </header>
    <div
      className="css-1xdhyk6"
    >
      <div
        className="css-0"
      >
        Notification content.
      </div>
    </div>
  </div>
  <button
    aria-label="Close the notification"
    className="chakra-button css-1xdhyk6"
    id="notificationID7-notification-dismissible-button"
    onClick={[Function]}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="chakra-icon css-13thq4x"
      focusable={false}
      id="notificationID7-dismissible-notification-icon"
      role="img"
      title="Notification close icon"
      viewBox="0 0 24 24"
    >
      <g
        stroke="currentColor"
        strokeWidth="1.5"
      >
        <path
          d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
          fill="none"
          strokeLinecap="round"
        />
        <path
          d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
          fill="currentColor"
          strokeLinecap="round"
        />
        <circle
          cx="12"
          cy="12"
          fill="none"
          r="11.25"
          strokeMiterlimit="10"
        />
      </g>
    </svg>
  </button>
</aside>
`;

exports[`Notification renders the UI snapshot correctly 8`] = `
<aside
  className="css-kle7zy"
  data-type="standard"
  id="chakra"
>
  <div
    className="css-0"
  >
    <header
      className="css-1xdhyk6"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon notification-icon css-q3eg3a"
        focusable={false}
        id="chakra-notification-icon"
        role="img"
        title="Notification standard icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      <h4
        className="chakra-heading css-1xdhyk6"
        id="chakra-heading"
      >
        Notification Heading
      </h4>
    </header>
    <div
      className="css-1xdhyk6"
    >
      <div
        className="css-0"
      >
        Notification content.
      </div>
    </div>
  </div>
</aside>
`;

exports[`Notification renders the UI snapshot correctly 9`] = `
<aside
  className="css-1xdhyk6"
  data-testid="props"
  data-type="standard"
  id="props"
>
  <div
    className="css-0"
  >
    <header
      className="css-1xdhyk6"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon notification-icon css-q3eg3a"
        focusable={false}
        id="props-notification-icon"
        role="img"
        title="Notification standard icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      <h4
        className="chakra-heading css-1xdhyk6"
        id="props-heading"
      >
        Notification Heading
      </h4>
    </header>
    <div
      className="css-1xdhyk6"
    >
      <div
        className="css-0"
      >
        Notification content.
      </div>
    </div>
  </div>
</aside>
`;
